src/Sparrow/Json/BlittableJsonReaderArray.cs

```diff
@@ -136,7 +136,13 @@ protected bool Equals(BlittableJsonReaderArray other)
 
             for (int i = 0; i < _count; i++)
             {
-                if (this[i].Equals(other[i]) == false)
+                var x = this[i];
+                var y = other[i];
+
+                if (x == null && y == null)
+                    continue;
+
+                if (x?.Equals(y) == false)
```

Comment:
You are correct. I was convinced that elvis operator works differently in conditional statements.